### PR TITLE
Modify profile provider

### LINKED ISSUE #362: Enhanced provider profile modification and fixed text field limits
- **Feature:** Enable providers to edit their profile information, including name, company name, service, phone number, location, and languages.
- **UI Improvement:** Set maximum line limits for text fields to enhance user experience and prevent UI issues.
- **Testing:** Added additional tests to verify the functionality and correctness of the changes.
- **Code Formatting:** Applied KTFMTFormat for consistent code style and readability.

### PR DESCRIPTION
**Pull request by ethanboren**
### Summary:
This pull request introduces new features and fixes to improve the profile modification functionality. After merging this PR,

### Changes:
- **Feature:** 
   - Enabled editing of the provider's name in the profile modification screen.
   - I modified the method of how to save provider data in its profile by better using the methods given in the viewmodel
   - I also modify a little bit the viewModel of the provider
- **Fix:** Set maximum lines for text fields in the profile to prevent UI breakage.
- **Style:** Improved the UI design of the provider profile and the profile modification screen to enhance aesthetics and user experience.
- Added comprehensive documentation for each method to improve code readability and maintainability.

### Additional Updates:
- **Tests:** Added comprehensive tests to ensure all functionalities work correctly.
- **Code Formatting:** Applied code formatting with KTFMTFormat for consistency and readability.

### How to Test:
1. Log in with a provider account.  
2. Navigate to the provider profile screen.  
3. Modify some data without saving and exit the screen:  
   - Verify that no changes have been applied.  
4. Return to the profile modification screen and make changes.  
5. Save the changes:  
   - Confirm that all modifications are reflected in the profile.  
   - Check the Firestore database to ensure that the updated data is correctly stored.

### Related Issues:
- Fixes #362

### CODE REVIEW

**Review by yshdb5 — APPROVED** [review]
### Overall Comment
This pull request brings valuable improvements to `ModifyProviderInformation.kt`, `ProviderProfile.kt`, and related tests. The refactoring enhances readability, maintainability, and functionality, particularly in the provider profile modification UI. Nice work!

### Strengths
- **Readability and Maintainability**: Clearer variable names and new state variables (`newName`, `newCompanyName`, `newService`) improve code clarity and flow.
- **UI Enhancements**: Adding `Modifier.testTag` for elements like the title and save button, along with improved `CustomOutlinedTextField` error handling, boosts accessibility and testability.
- **Data Integrity**: Making `Provider` fields immutable and using `copy` for updates reduces unintended side effects.
- **User Experience**: The `LanguageList` component now toggles between showing all or a limited set of languages, improving usability.
- **Testing**: Updated tests reflect new UI elements and error handling, ensuring robustness.

### Areas for Improvement
- **UI consistency**: In `ModifyProviderInformation.kt`,  the fields Select Service and Select languages have a weird bottom bar which make them stand out from the other fields of the screen. Can you modify this before merging?
- **Code documentation**: Try adding documentation to your code, headers to the functions and composable you create so that future developers can understand better what each of them is used for.

### Conclusion
This PR enhances the codebase with improved readability, maintainability, and functionality. The updates are well-executed and thoroughly tested. Great job! Keep it up!

File: app/src/main/java/com/android/solvit/shared/model/provider/Provider.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/shared/model/provider/Provider.kt:1]: nice!

**Review by manunam2002 — APPROVED** [review]
### feedback

Nice job making the provider profile modification UI is completed with the ability to modify the name. Also it is nice that you made the code more readable with tags and variable renaming and the logic clearer by making the provider data class immutable.

**Comment by ethanboren** [comment]
Thanks for your insightful comments @yshdb5. I have modified what you advised me.

**Pull request merged**